distribution/metadata/v2_metadata_service.go

```diff
@@ -151,6 +158,9 @@ func (serv *v2MetadataService) GetDiffID(dgst digest.Digest) (layer.DiffID, erro
 // Add associates metadata with a layer DiffID. If too many metadata entries are
 // present, the oldest one is dropped.
 func (serv *v2MetadataService) Add(diffID layer.DiffID, metadata V2Metadata) error {
+	if serv.store == nil {
+		return nil
```

Comment:
I will add a comment here. I think the better solution is just to have an inmemory implementation or allow the pullers/pushers to have this service be nil. In the end we probably just want to support this interface backed by disk because it can optimize pushing.